packages/ajax/test/AjaxClass.interceptors-cache.test.js

```diff
@@ -0,0 +1,713 @@
+      })
+      .then(() => ajax.get('/test'))
+      .then(() => {
+        expect(server.requests.length).to.equal(1);
+      })
+      .finally(() => {
+        ajaxGetSpy.restore();
+        removeCacheInterceptors(ajax, indexes);
+      });
+  });
+
+  it('all calls with non-default `timeToLive` are cached proactively', () => {
+    newCacheId();
+
+    const indexes = addCacheInterceptors(ajax, {
+      useCache: 'never',
+      timeToLive: 100,
+    });
+    const ajaxGetSpy = sinon.spy(ajax, 'get');
+
+    return ajax
+      .get('/test')
+      .then(() => {
+        expect(ajaxGetSpy.calledOnce).to.be.true;
+        expect(ajaxGetSpy.calledWith('/test')).to.be.true;
+      })
+      .then(() => {
+        expect(server.requests.length).to.equal(1);
+      })
+      .then(() => ajax.get('/test'))
+      .then(() => {
+        expect(server.requests.length).to.equal(2);
+      })
+      .then(() =>
+        ajax.get('/test', {
+          lionCacheOptions: {
+            useCache: 'always',
+          },
+        }),
+      )
+      .then(() => {
+        expect(server.requests.length).to.equal(2);
+      })
+      .finally(() => {
+        ajaxGetSpy.restore();
+        removeCacheInterceptors(ajax, indexes);
+      });
+  });
+
+  it('previously cached data has to be invalidated when regex invalidation rule triggered', () => {
+    newCacheId();
+
+    const indexes = addCacheInterceptors(ajax, {
+      useCache: 'always',
+      timeToLive: 1000,
+    });
+
+    server.respondWith(/\/foo-request-1/, xhr => {
+      xhr.respond(200, { 'Content-Type': 'application/json' }, '{}');
+    });
+
+    server.respondWith(/\/foo-request-2/, xhr => {
+      xhr.respond(200, { 'Content-Type': 'application/json' }, '{}');
+    });
+
+    const actionConfig = {
+      lionCacheOptions: {
+        invalidateUrlsRegex: /foo/gi,
+      },
+    };
+
+    return ajax
+      .get('/test', actionConfig)
+      .then(() => ajax.get('/test'))
+      .then(() => {
+        expect(server.requests.length).to.equal(1);
+      })
+      .then(() => ajax.get('/foo-request-1'))
+      .then(() => {
+        expect(server.requests.length).to.equal(2);
+      })
+      .then(() => ajax.get('/foo-request-1'))
+      .then(() => {
+        expect(server.requests.length).to.equal(2);
+      })
+      .then(() => ajax.get('/foo-request-2'))
+      .then(() => {
+        expect(server.requests.length).to.equal(3);
+      })
+      .then(() => ajax.get('/foo-request-2'))
+      .then(() => {
+        expect(server.requests.length).to.equal(3);
+      })
+      .then(() => ajax.post('/test', {}, actionConfig))
+      .then(() => {
+        expect(server.requests.length).to.equal(4);
+      })
+      .then(() => ajax.get('/foo-request-1'))
+      .then(() => {
+        expect(server.requests.length).to.equal(5);
+      })
+      .then(() => ajax.get('/foo-request-2'))
+      .then(() => {
+        expect(server.requests.length).to.equal(5);
```

Comment:
I would expect 6 here because the `ajax.post('/test', {}, actionConfig)` has the invalidate regex rule, which invalidates cache right?

Does it only invalidate "global" cache for the next request to "any" url? Meaning that the foo-request-2 (which is a different url) gets you a cached response immediately?

In general, I guess I don't understand, why do we have to invalidate cache when invalidation rules change?